articles/evaluating-ml-model-performance/index.md

```diff
@@ -0,0 +1,157 @@
+The model will be trained and tested four times. Let’s say the first-round trains on folds 1,2 and 3. The testing will be on fold 4. Fort the second round, it may train on folds 1,2, and 4 and test on fold 3. For the third, it may train on folds 1,3, and 4 and test on fold 2. The last round will test on folds 2,3 and 4 and test on fold 1.
+The interchange between training and test data makes this method very effective. However, compared to the holdout technique, cross-validation takes more time to run and uses more computational resources.
+
+#### Holdout
+
+It is important to get an unbiased estimate of model performance. This is exactly what the holdout technique offers. To get this unbiased estimate, a model is tested on data that is different from the data it was trained on.
+This technique divides a dataset into three subsets; training, validation, and test sets. From the terms we defined at the start of the article, we know that the training set helps the model make predictions and that the test set assesses the performance of the model. The validation set also helps to assess the performance of the model by providing an environment to fine-tune the parameters of the model. From this, the best performing model is selected.
+The holdout method is ideal for use when dealing with a very large dataset or when working with really limited time. The holdout method is faster than cross-validation, prevents the model from overfitting, and incurs lower computational costs.
+
+### Model Evaluation Metrics
+
+#### Metrics for Classification Problems
+
+Predictions for classification problems yield four types of outcomes; true positives, true negatives, false positives, and false negatives. We shall define them later on. We look at a few metrics for classification problems.
+
+##### Classification Accuracy
+
+The most common evaluation metric for classification problems is accuracy. It is taken as the number of correct predictions against the total number of predictions made (or input samples). However, as much as accuracy is used to evaluate a model, it is not a clear indicator of model performance.
+Classification accuracy works best if the samples belonging to each class are equal in number. Consider a scenario with 97% samples of class X and 3% class Y in a training set. A model can very easily achieve 97% training accuracy by predicting each training sample in class X.
+Testing the same model on a test set with 55% samples of X and 45% samples of Y, the test accuracy is reduced to 55%. This is why classification accuracy is not a clear indicator of performance. It provides a false sense of attaining high levels of accuracy.
+
+##### Confusion Matrix
+
+The confusion matrix forms the basis for the other types of classification metrics. It is a matrix that fully describes the performance of the model. A confusion matrix gives an in-depth breakdown of the correct and incorrect classifications of each class.
```

Comment:
In a confusion matrix, what values are acceptable? Is it alright to have false positives? Is it alright to have true negatives? Which ones should be minimized? What insights does one get from CM? Answer these questions. Moreover, define precision and recall intuitively. In terms of the memory property of the model.